unchanged:
--- hpl-2.0_FERMI_v15/src/cuda/cuda_dgemm.c	2017-01-13 14:45:54.411525307 -0800
+++ hpl-2.0_FERMI_v15/src/cuda/cuda_dgemm_modified.c	2017-01-13 14:49:01.262022385 -0800
@@ -360,12 +360,17 @@
           cudaSetDevice(mydev);
 #ifdef GOTO
       handle2 = dlopen ("libgoto2.so", RTLD_LAZY);
-#endif
+#else
+#ifdef OPENBLAS
+      handle2 = dlopen("libopenblas.so", RTLD_LAZY);
+#else
 #ifdef ACML
       handle2 = dlopen ("libacml_mp.so", RTLD_LAZY);
 #else
       handle2 = dlopen ("libmkl_intel_lp64.so", RTLD_LAZY);
 #endif
+#endif
+#endif
 
       if (!handle2) 
       {
@@ -376,12 +381,17 @@
 
 #ifdef GOTO
       mkl_dtrsm = (void(*)())dlsym(handle2, "dtrsm_");
-#endif
+#else
+#ifdef OPENBLAS
+      mkl_dtrsm = (void(*)())dlsym(handle2, "dtrsm_");
+#else
 #ifdef ACML
       mkl_dtrsm = (void(*)())dlsym(handle2, "dtrsm_");
 #else
       mkl_dtrsm = (void(*)())dlsym(handle2, "dtrsm");
 #endif
+#endif
+#endif
 
       if ((error = dlerror()) != NULL)  {
           fprintf (stderr, "%s\n", error);
@@ -654,12 +664,17 @@
           cudaSetDevice(mydev);
 #ifdef GOTO
       handle = dlopen ("libgoto2.so", RTLD_LAZY);
-#endif
+#else
+#ifdef OPENBLAS
+      handle = dlopen("libopenblas.so", RTLD_LAZY);
+#else
 #ifdef ACML
       handle = dlopen ("libacml_mp.so", RTLD_LAZY);
 #else
       handle = dlopen ("libmkl_intel_lp64.so", RTLD_LAZY);
 #endif
+#endif
+#endif
 
       if (!handle) {
         fprintf (stderr, "%s\n", dlerror());
@@ -669,12 +684,17 @@

 #ifdef GOTO
       dgemm_mkl = (void(*)())dlsym(handle, "dgemm_");
-#endif
+#else
+#ifdef OPENBLAS
+      dgemm_mkl = (void(*)())dlsym(handle, "dgemm_");
+#else
 #ifdef ACML
       dgemm_mkl = (void(*)())dlsym(handle, "dgemm_");
 #else
       dgemm_mkl = (void(*)())dlsym(handle, "dgemm");
 #endif
+#endif
+#endif


        if ((error = dlerror()) != NULL)  {
only in patch2:
unchanged:
--- hpl-2.0_FERMI_v15/Make.CUDA	2012-07-16 19:35:11.000000000 -0700
+++ hpl-2.0_FERMI_v15/Make.CUDA.updated	2017-01-13 13:32:27.330821564 -0800
@@ -133,7 +133,7 @@
 LAinc        =
 # CUDA
 #LAlib        = -L /home/cuda/Fortran_Cuda_Blas  -ldgemm -L/usr/local/cuda/lib -lcublas  -L$(LAdir) -lmkl -lguide -lpthread
-LAlib        = -L $(TOPdir)/src/cuda  -ldgemm -L/usr/local/cuda/lib64 -lcuda -lcudart -lcublas -L$(LAdir) -lmkl_intel_lp64 -lmkl_intel_thread -lmkl_core -liomp5
+LAlib        = -L $(TOPdir)/src/cuda  -ldgemm -L/usr/local/cuda/lib64 -lcuda -lcudart -lcublas -lopenblas -lpthread -lgfortran
 #
 # ----------------------------------------------------------------------
 # - F77 / C interface --------------------------------------------------
--- hpl-2.0_FERMI_v15/src/cuda/Makefile    2012-07-17 02:35:11.000000000 +0000
+++ hpl-2.0_FERMI_v15/src/cuda/Makefile.modified   2017-01-14 00:42:27.078695198 +0000
@@ -39,6 +39,7 @@
 #DEFINES += -DVERBOSE_PRINT
 #DEFINES += -DACML
 #DEFINES += -DGOTO
+DEFINES += -DOPENBLAS

 %.o: %.c
        mpicc -O0 -c -fPIC $(DEFINES) $*.c -o $*.o -I/usr/local/cuda/include

